### DISCUSSION TITLE
Can't convert dates to Firestore timestamps

### DISCUSSION THREAD

**Discussion by gospodinove** [invertase/react-native-firebase#5966] [Q&A] [answered]
I'm writing a TypeScript project and I have some date (timestamp) fields in my firestore DB. 

I am storing them as firestore timestamps but I convert them to dates with the provided `toDate` method. However, I don't know how to convert date objects to timestamps in order to send them to the DB.

I tried using `FirebaseFirestoreTypes.Timestamp.fromDate(date)`, but then I get this error
![Simulator Screen Shot - iPhone 13 - 2021-12-28 at 01 54 41](https://user-images.githubusercontent.com/5853546/147514379-c354690a-f2f4-4b7d-8f96-0108706c0ad0.png)

Any help is welcome :) I looked it up online but nothing came up so I'm turning to you guys here

Thanks in advance 🙏🏻

**Comment by andersonaddo** [comment, marked as answer]
I think here's a right way to use it:
https://github.com/invertase/react-native-firebase/blob/8efda506474b1757f6a0537495a1de581d372299/packages/firestore/e2e/DocumentSnapshot/data.e2e.js#L71

-----


Looking at 
https://github.com/invertase/react-native-firebase/blob/8efda506474b1757f6a0537495a1de581d372299/packages/firestore/lib/index.js#L337-L348

which points to 
https://github.com/invertase/react-native-firebase/blob/8efda506474b1757f6a0537495a1de581d372299/packages/firestore/lib/FirestoreStatics.js#L25-L30

**Reply by gospodinove** [reply]
Thank you! It was just what I needed. I suppose the error came from me using `FirebaseFirestoreTypes`, which I guess is meant to help support TypeScript. Do you know if this is correct?
